### DISCUSSION TITLE
making yolo v8 train on data that have it directory listed on .csv

### DISCUSSION THREAD

**Discussion by Mohamed-Elesawy** [ultralytics/ultralytics#3834] [Q&A]
i want to know how can i give yolo v8  a .csv file that include the paths for the training and validation image for an object detection task instead of the root folder directory.

**Comment by glenn-jocher** [comment]
@mohaMKRaod hello! To train YOLOv8 with a `.csv` file containing the paths to your training and validation images, you'll need to convert your `.csv` data into the format that YOLOv8 expects for training. YOLOv8 uses a specific directory structure and annotation format, typically with a `.yaml` file that specifies the paths to your dataset directories and classes.

Here's a brief overview of the steps you would follow:

1. **Convert CSV to YOLO Format**: You'll need to convert your `.csv` file into two separate text files for training and validation, where each line contains the path to an image followed by the bounding box annotations in YOLO format.

2. **Create Dataset Directory**: Organize your images and annotation text files into the appropriate directory structure. For example, you might have a `dataset` folder with subfolders `images` and `labels` for both training and validation.

3. **Prepare YAML File**: Create a `.yaml` file that lists the paths to your training and validation directories, as well as the class names.

4. **Train**: Use the YOLOv8 CLI or Python environment to start training with the `.yaml` file.

For detailed instructions on the expected dataset format and the training process, please refer to the documentation at https://docs.ultralytics.com. The documentation provides comprehensive guidance on preparing your data and configuring your training environment. Happy training! 🚀
